internal/pkg/cli/pipeline_package.go

```diff
@@ -0,0 +1,220 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cli
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"path/filepath"
+
+	"github.com/aws/copilot-cli/internal/pkg/cli/list"
+	"github.com/aws/copilot-cli/internal/pkg/config"
+	"github.com/aws/copilot-cli/internal/pkg/deploy"
+	"github.com/aws/copilot-cli/internal/pkg/manifest"
+)
+
+type packagePipelineVars struct {
+	name    string
+	appName string
+}
+
+type packagePipelineOpts struct {
+	packagePipelineVars
+
+	pipelineDeployer                pipelineDeployer
+	tmplWriter                      io.WriteCloser
+	ws                              wsPipelineReader
+	codestar                        codestar
+	store                           store
+	pipelineStackConfig             func(in *deploy.CreatePipelineInput) pipelineStackConfig
+	configureDeployedPipelineLister func() deployedPipelineLister
+	newSvcListCmd                   func(io.Writer, string) cmd
+	newJobListCmd                   func(io.Writer, string) cmd
+
+	//catched variables
+	pipelineMft *manifest.Pipeline
+	app         *config.Application
+	svcBuffer   *bytes.Buffer
+	jobBuffer   *bytes.Buffer
+}
+
+func (o *packagePipelineOpts) Execute() error {
```

Comment:
i wonder if we should do https://github.com/aws/copilot-cli/pull/4991#discussion_r1231599761 as part of this PR? seems like most of this code is duplicated from `pipeline_deploy.go` and we should combine in `cli/deploy`

Reply:
chatted with @iamhopaul123 offline - let's save the refactor until after the `override` feature is merged!